test/jdk/sun/security/pkcs11/SecureRandom/Basic.java

```diff
@@ -44,9 +46,8 @@ public void main(Provider p) throws Exception {
         try {
             random = SecureRandom.getInstance("PKCS11");
         } catch (NoSuchAlgorithmException e) {
-            System.out.println("Provider " + p + " does not support SecureRandom, skipping");
             e.printStackTrace();
-            return;
+            throw new SkippedException("Provider " + p + " does not support SecureRandom, skipping");
```

Comment:
If you add `e` as the cause here, you can probably get rid of the explicit `e.printStackTrace()`